// @nuxt/vue-app/dist/vue-app.js
/*!
 * @nuxt/vue-app v2.12.0 (c) 2016-2020

 * - All the amazing contributors
 * Released under the MIT License.
 * Website: https://nuxtjs.org
*/
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

const path = _interopDefault(require('path'));

var dependencies = {
	"node-fetch": "^2.6.0",
	unfetch: "^4.1.0",
	vue: "^2.6.11",
	"vue-client-only": "^2.0.0",
	"vue-meta": "^2.3.3",
	"vue-no-ssr": "^1.1.1",
	"vue-router": "^3.1.6",
	"vue-template-compiler": "^2.6.11",
	vuex: "^3.1.3"
};

const template = {
  dependencies,
  dir: path.join(__dirname, '..', 'template'),
  files: [
    'App.js',
    'client.js',
    'index.js',
    'router.js',
    'router.scrollBehavior.js',
    'server.js',
    'utils.js',
    'empty.js',
    'mixins/fetch.server.js',
    'mixins/fetch.client.js',
    'components/nuxt-error.vue',
    'components/nuxt-child.js',
    'components/nuxt-link.server.js',
    'components/nuxt-link.client.js',
    'components/nuxt.js',
    'views/app.template.html',
    'views/error.html'
  ]
};

exports.template = template;
